### DISCUSSION TITLE
테라폼코드이용하여미리준비된AWS실습환경구성 질문이있습니다!

### DISCUSSION THREAD

**Discussion by eungkyoung** [tedilabs/fastcampus-devops#86] [Q&A] [answered]
안녕하세요! 패스트캠퍼스에서 Kubernetes 와 Docker로 한 번에 끝내는 컨테이너 기반 MSA 강의를 결제하여 수강하고 있는 학생입니다.
입문 강좌로 알고 결제 했는데, 윈도우를 이용하고 있어서 aws실습환경 구성하는 법이 이해가 되지 않습니다.. ㅠㅠ aws를 사용해 본 적이 없어서 그런데, 혹시 실습환경 구성하는데 참고할만한 다른 강의나 글이 있을까요?

**Comment by posquit0** [comment, marked as answer]
@eungkyoung 안녕하세요! 이슈를 겪고 있으시군요.. 😢 

<img width="663" alt="image" src="https://user-images.githubusercontent.com/1484002/217266346-46ba509e-4b06-414f-b4ca-239555e17da1.png">

<Kubernetes와 Docker로 한 번에 끝내는 컨테이너 기반 MSA 강의>의 도커 기초 파트는 패스트캠퍼스 측에서 <데브옵스 초격차> 강의 일부를 발췌한 것으로 전체 강의 내용이 서로 다르다보니 문제가 발생한 것으로 보입니다..

우선 도커와 쿠버네티스를 공부하실 예정이라면 윈도우 환경은 추천하지 않습니다. 요새는 윈도우에서 WSL 기능을 이용하여 윈도우 내에서 Ubuntu를 이용하는 것도 가능하나, 도커 컨테이너를 이용하게 될 경우 실제 Ubuntu와는 다른 점들을 경험하게 됩니다.
컨테이너 가상화 기술은 리눅스에 기반을 둔 기술이기 때문에 학습 시에도 리눅스 기반으로 진행하시는 것이 다른 문제를 겪지 않고 학습에 집중하실 수 있습니다. Windows나 macOS에서도 컨테이너 기술을 활용할 수 있도록 여러 도구들(Docker for Desktop 등)이 나와있지만, 강의에서 언급하는 내용처럼 네트워크 혹은 볼륨 설정 등을 다루다보면 제약사항들과 마주하게 됩니다.

실습 환경을 구성하기 위한 방법은 여러가지가 있는데, 가장 편한 방법으로 진행하시면 될 것 같습니다.

- 윈도우에 VMware / VirtualBox 등 가상머신 소프트웨어를 이용하여 Ubuntu 가상머신 생성
  - https://mainia.tistory.com/2379
- 남는 노트북이나 PC 등에 Ubuntu 운영체제 설치
  - https://wjdtmddnjs6788.tistory.com/941
- AWS 와 같은 퍼블릭 클라우드를 이용하여 Ubuntu 기반 서버 생성
  - https://velog.io/@newon-seoul/AWS-EC2%EC%97%90-%EB%A6%AC%EB%88%85%EC%8A%A4-%EA%B8%B0%EB%B0%98%EC%9A%B0%EB%B6%84%ED%88%AC-20.04LTS-%EC%9D%B8%EC%8A%A4%ED%84%B4%EC%8A%A4-%EC%83%9D%EC%84%B1%ED%95%98%EA%B8%B0

문제 해결에 도움이 되었으면 좋겠네요. 😭 
감사합니다!
